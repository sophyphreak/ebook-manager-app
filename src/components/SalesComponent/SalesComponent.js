import React from 'react';
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  FormGroup,
  Input,
  Label,
  Row
} from 'reactstrap';

import BasicInput from '../FormElements/MainElements/BasicInput';

const SalesComponent = ({
  bsr,
  sales,
  onBsrChange
}) => (
  <Row className="animated fadeIn">
    <Col xs="12" sm={{ size: 6, offset: 3 }}>
      <Card>
        <CardHeader>
          <h4><strong>Sales Estimator</strong></h4>
        </CardHeader>
        <CardBody>
          <BasicInput
            label="Enter BSR"
            onChange={onBsrChange}
            value={bsr}
          />
          <br />
          {
            bsr &&
            <FormGroup className="animated fadeIn">
              <Label>Estimated sales</Label>
              <br /> 
              <Label><b>{sales} units per month</b></Label>
            </FormGroup>
          }
        </CardBody>
      </Card>
    </Col>
  </Row>
);

export default SalesComponent;