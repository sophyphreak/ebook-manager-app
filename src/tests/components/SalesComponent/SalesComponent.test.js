import React from 'react';
import { shallow } from 'enzyme';
import SalesComponent from '../../../components/SalesComponent/SalesComponent';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<SalesComponent />);
});

test('should render SalesComponent correctly', () => {
  expect(wrapper).toMatchSnapshot();
});