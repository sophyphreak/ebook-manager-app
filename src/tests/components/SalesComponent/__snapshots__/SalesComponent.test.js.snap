// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SalesComponent correctly 1`] = `
<Row
  className="animated fadeIn"
  tag="div"
>
  <Col
    sm={
      Object {
        "offset": 3,
        "size": 6,
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        <h4>
          <strong>
            Sales Estimator
          </strong>
        </h4>
      </CardHeader>
      <CardBody
        tag="div"
      >
        <BasicInput
          label="Enter BSR"
        />
        <br />
      </CardBody>
    </Card>
  </Col>
</Row>
`;
